### PR TITLE
pcapplusplus 18.08 (new formula)

### PR DESCRIPTION
**Pull request by seladb**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/pcapplusplus.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Pcapplusplus < Formula
+  desc "Multiplatform C++ network sniffing and packet parsing and crafting"
+  homepage "https://seladb.github.io/PcapPlusPlus-Doc"
+  url "https://github.com/seladb/PcapPlusPlus/archive/v18.08.tar.gz"
+  sha256 "dff6f7c677b2050f880043b125e984238cd8af0f1c25864e09e87fb8d71ec9ab"
+
+  def install
+    system "./configure-mac_os_x.sh", "--install-dir", prefix.to_s
+
+    # library requires to run 'make all' and 'make install' in two sepearate commands
```
Comment (DomT4) [inline Formula/pcapplusplus.rb]: Minor typo: `separate` instead of `sepearate`. Not a huge thing.

File: Formula/pcapplusplus.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class Pcapplusplus < Formula
+  desc "Multiplatform C++ network sniffing and packet parsing and crafting"
+  homepage "https://seladb.github.io/PcapPlusPlus-Doc"
+  url "https://github.com/seladb/PcapPlusPlus/archive/v18.08.tar.gz"
+  sha256 "dff6f7c677b2050f880043b125e984238cd8af0f1c25864e09e87fb8d71ec9ab"
+
+  def install
+    system "./configure-mac_os_x.sh", "--install-dir", prefix.to_s
```
Comment (DomT4) [inline Formula/pcapplusplus.rb]: The `.to_s` should be unnecessary.

File: Formula/pcapplusplus.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      int main() {
+        pcpp::PcapLiveDevice* dev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByName("en0");
+        if (dev == NULL)
+          exit(1);
+        exit(0);
+      }
+    EOS
+
+    (testpath/"Makefile").write <<~EOS
+      include /usr/local/etc/PcapPlusPlus.mk
```
Comment (DomT4) [inline Formula/pcapplusplus.rb]: Don't assume `HOMEBREW_PREFIX` will always == `/usr/local`. Use `#{etc}` instead.

**Review by DomT4 — DISMISSED** [review]

**Comment by seladb** [comment]
Thanks for the feedback! I fixed the issues you called out, please review and let me know if more changes are needed

File: Formula/pcapplusplus.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class Pcapplusplus < Formula
+  desc "Multiplatform C++ network sniffing and packet parsing and crafting"
+  homepage "https://seladb.github.io/PcapPlusPlus-Doc"
+  url "https://github.com/seladb/PcapPlusPlus/archive/v18.08.tar.gz"
+  sha256 "dff6f7c677b2050f880043b125e984238cd8af0f1c25864e09e87fb8d71ec9ab"
+
+  def install
+    system "./configure-mac_os_x.sh", "--install-dir", prefix
+
+    # library requires to run 'make all' and 'make install' in two separate commands
```
Comment (DomT4) [inline Formula/pcapplusplus.rb]: Can we nudge this across two lines. Almost everything should be wrapped to 80 characters. Everything ideally, but obviously not always practical; comments are easy to wrap to a character limit though.

File: Formula/pcapplusplus.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    EOS
+
+    (testpath/"Makefile").write <<~EOS
+      include #{etc}/PcapPlusPlus.mk
+      all:
+      \tg++ $(PCAPPP_BUILD_FLAGS) $(PCAPPP_INCLUDES) -c -o test.o test.cpp
+      \tg++ $(PCAPPP_LIBS_DIR) -o test test.o $(PCAPPP_LIBS)
+    EOS
+
+    system "make", "all"
```
Comment (DomT4) [inline Formula/pcapplusplus.rb]: If `test` is being created can we execute it with `system "./test"` or is that going to fail for whatever expected/normal reason?

File: Formula/pcapplusplus.rb (hunk 4 above)
Comment (seladb) [reply Formula/pcapplusplus.rb]: Done

File: Formula/pcapplusplus.rb (hunk 5 above)
Comment (seladb) [reply Formula/pcapplusplus.rb]: Changed the test so it should pass if everything is ok and added `system "./test"` to run it

**Comment by seladb** [comment]
@DomT4 - issues are fixed, please let me know if you have more comments

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/b4cdc6c81a9af0725a00469a05cdafefcee9b219. Thank you for your contribution to Homebrew @seladb; we appreciate it! :smiley_cat:

**Comment by seladb** [comment]
Thank you @DomT4 and Homebrew team for including PcapPlusPlus in Homebrew! This will definitely improve PcapPlusPlus usability and installation process in Mac and Linux. Much appreciated!

**Pull request closed without merging**